### DISCUSSION TITLE
corosync/pacemaker ha

### DISCUSSION THREAD

**Discussion by 3157696** [strongswan/strongswan#1447] [Q&A]
Hello
We want to using ha through corosync/pacemaker
Is it possible?

If yes
What is the name of the tunnel  ? <conn> ?

[https://www.mankier.com/7/ocf_heartbeat_ipsec](url)

Thanks.

**Comment by misch42** [comment]
Hi,

Pacemaker is a really complex cluster management. Perhaps you try first with a simple VRRP like the keepalived service?

Michael Schwartzkopff

**Reply by 3157696** [reply]
Thank you for your reply
My 
os is debian: 11.5/strongswan:5.9.8

Because I tried this method of  ha/active-passive before unsuccess ,  I'm having problems，

`iptables -A INPUT -d 192.168.0.1/32 -i eth1 -j CLUSTERIP --new --hashmode sourceip --clustermac 01:00:C0:A8:00:05 --total-nodes 1 --local-node 0 --hash-init 0`
1:log : not support CLUSTERIP
2:clustermac 01:00:C0:A8:00:05  is this mac universal ？

https://docs.strongswan.org/docs/5.9/features/highAvailability.html#_failover

**Reply by tobiasbrunner** [reply]
While the active-passive scenario could work without special kernel patches, it won't work if CLUSTERIP isn't available at all (maybe it can be installed as module, but since it's been deprecated for a while and only supports IPv4 it's definitely possible that the Debian kernel has it disabled).

> clustermac 01:00:C0:A8:00:05 is this mac universal ？

Not sure what you mean exactly. It's simply an arbitrary MAC address that both nodes will use.

**Comment by misch42** [comment]
Again. Please use a simple VRRP failover system like keepalived.

clusterip ist really complex and you have to have deep understanding of networking to be able to apply it. clusterip is a method for load distribution over multiple nodes, it is no active/passive solution.

So please use keepalived.

**Reply by tobiasbrunner** [reply]
> clusterip is a method for load distribution over multiple nodes, it is no active/passive solution.

It can be, see [this test scenario](https://strongswan.org/testing/testresults/ha/active-passive/index.html).

**Reply by 3157696** [reply]
> Again. Please use a simple VRRP failover system like keepalived.
> 
> clusterip ist really complex and you have to have deep understanding of networking to be able to apply it. clusterip is a method for load distribution over multiple nodes, it is no active/passive solution.
> 
> So please use keepalived.

OK,I will try to use keepalived
Is it possible to share the configuration for reference? :)
